[신수녕] 화상연결 스크린 레이아웃 완성

cozynye opened the pull request:
### 최근 작업 주제

- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ]  컨벤션 수정

### 구현 목표
화상연결 스크린 레이아웃 구현

### 구현 사항 설명 
- react-native-webrtc를 활용한 레이아웃 구현
- 카메라 on, off 구현
- 카메라 전면, 후면 카메라 전환 구현

### 기타 질문 및 특이 사항
- 현재 오디오가 되도록 코드는 짠거 같은데  테스트가 필요합니다

cozynye committed:
Add : 화상연결 스크린 레이아웃 완성

Enjoywater commented on src/screens/VideoChat.tsx:
```diff
@@ -0,0 +1,234 @@
+      });
+  };
+
+  useEffect(() => {
+    initVideo();
+  }, []);
+
+  useEffect(() => {
+    initVideo();
+  }, [isFrontVideo, isVideo]);
```
`dependency`가 있어도 `mount`될 때 `hook`이 동작합니다.

cozynye replied:
제가 현재 원하는 기능이 카메라 전환버튼, 카메라 on,off 버튼 클릭시 initVideo() 함수를 실행하기 위해 위처럼 작성하였는데

 useEffect(() => {
    initVideo();
  }, [isFrontVideo, isVideo]);
-> 위 소스가 있기 때문에 마운트 될때 어차피 InitVideo()가 실행이 되니 

useEffect(() => {
    initVideo();
  }, []);
이 소스가 굳이 필요없다는 말씀이신가요?

Enjoywater replied:
네 맞습니다
```
 useEffect(() => {
    initVideo();
  }, []);
```
해당 hook은 처음에만 동작하고

```
useEffect(() => {
    initVideo();
  }, [isFrontVideo, isVideo]);
```
해당 hook은 처음 + 특정 값이 바뀔 때 동작하기 때문입니다

cozynye replied:
감사합니다!

Enjoywater commented on src/screens/VideoChat.tsx:
```diff
@@ -0,0 +1,234 @@
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+  padding-left: 10px;
+  padding-right: 10px;
+  padding-top: 30px;
+  padding-bottom: 64px;
```
단축속성으로 표현이 가능합니다

Enjoywater commented on src/screens/VideoChat.tsx:
```diff
@@ -0,0 +1,234 @@
+  bottom: 0;
+  width: 100%;
+  height: 32px;
+
```
여백은 제거해주세요

Enjoywater commented on src/screens/VideoChat.tsx:
```diff
@@ -0,0 +1,234 @@
+import CameraImage from 'assets/images/camera_revers_btn.png';
+import AudioOnImage from 'assets/images/audio_on_btn.png';
+import VideoOnImage from 'assets/images/video_on_btn.png';
+import VideoOffImage from 'assets/images/video_off_btn.png';
```
`assets` 폴더에 `index` 파일을 만들고, `index` 파일에서 이미지를 `export`하면 
`import {} from 'assets/images'` 로 여러개의 이미지를 불러올 수 있습니다

cozynye committed:
Fix : PR#5 comment resolve

jasonkang14 commented on ios/WebRTC/Info.plist:
```diff
@@ -51,6 +51,9 @@
 	<false/>
-	
+	<key>NSCameraUsageDescription</key>
+	<string>Camera permission description</string>
```
String에는 마이크 권한이 왜 필요한지 작성해야합니다. 아래 카메라도 마찬가지로 왜 카메라 권한이 필요한지 작성해야 합니다

cozynye replied:
\<key>NSCameraUsageDescription\</key>
	\<string>The purpose of non-face-to-face treatment\</string>
	\<key>NSMicrophoneUsageDescription\</key>
	\<string>The purpose of non-face-to-face treatment\</string>
이렇게 수정을 해도 괜찮을까요?

jasonkang14 replied:
[블로그](https://techcrunch.com/2014/04/04/the-right-way-to-ask-users-for-ios-permissions/)를 참고해서 영어로 작성해도 되고, 우리말로 작성해도 됩니다

Enjoywater commented on src/assets/images/index.ts:
```diff
@@ -0,0 +1,11 @@
+import VideoOnImage from './video_on_btn.png';
+import VideoOffImage from './video_off_btn.png';
+
+export const imageFile = {
+  CameraImage,
+  AudioOnImage,
+  VideoOnImage,
+  VideoOffImage,
+};
```
index파일로 합쳐주셨군요👍

위 경우에는 객체에 담아서 export하지 않고
`export { 이미지파일1, 이미지파일2, ...  }` 로도 사용이 가능하고

import를 사용하지 않고 한번에 export하려면
`export { default as 이미지파일1 } from './이미지파일1'` 로 사용이 가능합니다.

jasonkang14 reviewed (changes requested):
서버 연결 전에 본인의 영상만 보이도록 작업 한거라면 매우 좋습니다 👍
Info.plist 에서 권한을 가져올 때 문구를 수정해주시고
폰 스크린샷 등 불필요한 이미지를 삭제해주세요

cozynye committed:
Fix: pr#5 comment resolve
1. 불필요한 이미지 삭제
2. 이미지파일 Index.ts 객체 안에 담아서 export 하는 것 수정
3. info.plist 카메라, 오디오 권한 문구 수정

cozynye committed:
Fix: 불필요한 이미지 삭제
